File: Backend/AssignmentManagementSystem.BusinessLogicLayer/Entities/Assignment.cs
using System;
using System.Collections.Generic;

namespace AssignmentManagementSystem.BusinessLogicLayer.Entities;

public partial class Assignment
{
    public Guid Id { get; set; }
    public Guid TeacherId { get; set; }
    public Guid SubjectId { get; set; }
    public Guid ClassId { get; set; }
    public string Title { get; set; } = null!;
    public string? Description { get; set; }
    public int? TotalMarks { get; set; }
    public DateTime? Duedate { get; set; }
    public string? AttachmentUrl { get; set; }
    public string? Status { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public bool IsActive { get; set; }

    public virtual Class Class { get; set; } = null!;
    public virtual Subject Subject { get; set; } = null!;
    public virtual ICollection<Submission> Submissions { get; set; } = new List<Submission>();
    public virtual User Teacher { get; set; } = null!;
}

File: Frontend/src/types/assignment.ts
// Frontend/src/types/assignment.ts

export interface Assignment {
    id: string;
    teacherId: string;
    teacherName: string;
    subjectId: string;
    subjectName: string;
    classId: string;
    className: string;
    title: string;
    description?: string;
    totalMarks?: number;
    dueDate?: string;
    status?: string;
    attachmentUrl?: string;
    isActive: boolean;
    createdAt: string;
    updatedAt?: string;
    daysRemaining?: number;
    isOverdue?: boolean;
}

export interface CreateAssignmentData {
    teacherId: string;
    subjectId: string;
    classId: string;
    title: string;
    description?: string;
    totalMarks?: number;
    dueDate?: string;
    status?: string;
    attachmentUrl?: string;
    isActive?: boolean;
}

export interface UpdateAssignmentData {
    id: string;
    subjectId: string;
    classId: string;
    title: string;
    description?: string;
    totalMarks?: number;
    dueDate?: string;
    status?: string;
    attachmentUrl?: string;
    isActive?: boolean;
}

export interface CreateResponse {
    success: boolean;
    message: string;
    data: Assignment;
}

export interface UpdateResponse {
    success: boolean;
    message: string;
    data: Assignment;
}

export interface DeleteResponse {
    success: boolean;
    message: string;
}